libs/moment-datetime-adapter/adapter/moment-datetime-adapter.ts

```diff
@@ -0,0 +1,312 @@
+        date = this.clone(date);
+
+        if (!this.isValid(date)) {
+            throw Error('MomentDateTimeAdapter: Cannot format invalid date.');
+        }
+
+        return date.format(displayFormat);
+    };
+
+    createDate(year: number, month: number, date: number): Moment {
+        if (month < 0 || month > 11) {
+            throw Error(`Invalid month index "${ month }". Month index has to be between 0 and 11.`);
+        }
+
+        if (date < 1) {
+            throw Error(`Invalid date "${ date }". Date has to be greater than 0.`);
+        }
+
+        const result = this._createMomentDate({ year: year, month: month, date: date }).locale(this.locale);
+
+        if (!result.isValid()) {
+            throw Error(`Invalid date "${ date }" for month with index "${ month }".`);
+        }
+
+        return result;
+    };
+
+    today(): Moment {
+        return this._createMomentDate()
+            .locale(this.locale)
+            .startOf('day');
+    };
+
+    now(): Moment {
+        return this._createMomentDate().locale(this.locale);
+    };
+
+    addCalendarYears(date: Moment, years: number): Moment {
+        return this.clone(date).add({ years: years });
+    };
+
+    addCalendarMonths(date: Moment, months: number): Moment {
+        return this.clone(date).add({ months: months });
+    };
+
+    addCalendarDays(date: Moment, days: number): Moment {
+        return this.clone(date).add({ days: days });
+    };
+
+    getAmountOfWeeks(year: number, month: number, firstDayOfWeek: number): number {
+        return null;
+    };
+
+    clone(date: Moment): Moment {
+        return date.clone().locale(this.locale);
+    };
+
+    datesEqual(date1: Moment, date2: Moment): boolean {
+        if (!date1 || !date2) {
+            return false;
+        }
+
+        return date1.isSame(date2, 'day');
+    };
+
+    dateTimesEqual(date1: Moment, date2: Moment): boolean {
+        if (!date1 || !date2) {
+            return false;
+        }
+
+        return date1.isSame(date2);
+    };
+
+    isBetween(dateToCheck: Moment, startDate: Moment, endDate: Moment): boolean {
+        if (!dateToCheck || !startDate || !endDate) {
+            return false;
+        }
+
+        return dateToCheck.isBetween(startDate, endDate);
+    };
+
+    isValid(date: Moment): boolean {
+        return this.clone(date).isValid();
+    };
+
+    toIso8601(date: Moment): string {
+        return this.clone(date).format();
+    };
+
+    isTimeFormatIncludesDayPeriod(displayFormat: string): boolean {
+        return !!displayFormat.match(/[aA]/);
+    };
+
+    isTimeFormatIncludesHours(displayFormat: string): boolean {
+        return !!displayFormat.match(/[hH]/);
+    };
+
+    isTimeFormatIncludesMinutes(displayFormat: string): boolean {
+        return !!displayFormat.match(/[m]/);
+    };
+
+    isTimeFormatIncludesSeconds(displayFormat: string): boolean {
+        return !!displayFormat.match(/[s]/);
+    };
```

Comment:
That wont work with localized formats (e.g. `LLLL`)

Reply:
Let's check if are dealing with localized format and convert it

![image](https://user-images.githubusercontent.com/3263744/107212569-d1180c80-6a0f-11eb-9d0d-1ade1f3130b8.png)

Reply:
Re-worked